eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -451,7 +451,12 @@ function RetrievePackages($pkgRepository, $artifactLocation) {
       exit(1)
     }
   }
-  $pkgs = Get-ChildItem -Path $artifactLocation -Include $packagePattern -Recurse -File
+  $file_regex = if (!$pkgName) {
```

Comment:
What prompted this change?

Reply:
I found js is having multiple packages under artifacts. We need to filter out the one we use.
Checked it here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=522770&view=artifacts&type=publishedArtifacts

Reply:
this isn't exactly a regex so the name is a little miss-leading. I would instead do something like:

```
if (!$pkgName) {
  $packagePattern = $pkgName + $packagePattern
}
```